Filter to add X-Frame-Options header and prevent clickjacking attacks

josegar74 opened the pull request.

josegar74 commented:
I can't add the documentation to the PR as docs is a submodule, but will add it when approved.

A test page example to test in a web server:

```
<html>
<body>
my frame:
<iframe src="http://localhost:8080/geonetwork/">
</body>
</html>
```

If the filter is disabled (commented), the frame content is loaded with the home page. 

If the filter is enabled, the frame content is not loaded and in the developer tools of Chrome for example can see the request with an error `ERR_BLOCKED_BY_RESPONSE`.

fxprunayre commented:
I've seen user setting this is up at the Apache reverse proxy level. BTW why not adding it also in GeoNetwork. What about the 3 options ?

```
X-Frame-Options: ALLOW-FROM https://example.com/
```

josegar74 commented:
I can add that option, but note that is not supported by all browsers. From https://www.owasp.org/index.php/Clickjacking_Defense_Cheat_Sheet:

```
ALLOW-FROM uri, which permits the specified 'uri' to frame this page. (e.g., ALLOW-FROM http://www.example.com) Check Limitations Below this will fail open if the browser does not support it. Other browsers support the new CSP frame-ancestors directive instead. A few support both.
```

I'll update the pull request.

josegar74 commented:
@fxprunayre, updated the pull request with `ALLOW-FROM` support. Added also support for `Content-Security-Policy frame-ancestors`, so Chrome for example that doesn't support `X-Frame-Options ALLOW-FROM` works also.

Please if you can check and approve if looks all fine.

josegar74 commented:
@Delawen I have made a PR for documentation related to this: https://github.com/geonetwork/doc/pull/24

@fxprunayre I have added the support for `ALLOW-FROM` mode

Please check now, I would not like to loose or have to redo the PR again in the future due to conflicts. Thanks.

Delawen commented:
@fxprunayre what do you think?
